tests/integration_tests.rs

```diff
@@ -122,9 +122,10 @@ fn immediate_persistence() {
     test_persistence(Durability::Immediate);
 }
 
+#[cfg(not(feature = "wasi"))] // TODO figure out why this fails on WASI
 #[test]
 fn free() {
```

Comment:
This is the only test that is failing that shouldn't be failing. Looked thru and not entirely sure what is going on in this test so might need some help to look at this:
```
test free ... thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `2`,
 right: `275`', tests/integration_tests.rs:189:5
```
Line in question:
```rust
    assert_eq!(allocated_pages, txn.stats().unwrap().allocated_pages());
```
Wondering if this is something to do with page size or the fact that WASM is 32 bit? Not entirely sure though.

Reply:
For reference, expected and actual are both `2` on my macOS computer. What could cause the actual value to be 275 in WASI?

Reply:
I bet this is a usize = 32bits bug. I've had a couple of those. I think the problem is line 162. Try changing that to `(i as u64).to_le_bytes()` so that it matches line 177

Reply:
yep, 99% sure that's the issue. Try with this patch: https://github.com/cberner/redb/pull/588

Reply:
Thanks, was wondering if it had to do with it being 32 bit. I’ll try it out tomorrow

Reply:
It's passing locally now, thanks!